### DISCUSSION TITLE
Run event after window close

### DISCUSSION THREAD

**Discussion by certainlyWrong** [flet-dev/flet#914] [General]
Hello everyone, I would like to know if it is possible to register an event to be executed after closing the window.

**Comment by FeodorFitsner** [comment]
Would checking for `close` on [`Page.on_window_event`](https://flet.dev/docs/controls/page#on_window_event) work?

**Reply by dd-rongfa** [reply]
Do Not Work

**Comment by ndonkoHenri** [comment]
Follow the example here: https://flet.dev/docs/controls/page/#window_destroy. Inside the `window_event` callback, register your said event in a thread, then explicitly call the `window_destroy`.

**Comment by certainlyWrong** [comment]
Thank you, this was exactly what I wanted.

**Reply by certainlyWrong** [reply]
Just to clarify, I didn't find anything talking about this in the Flet documentation. Is the documentation out of date or did I miss it?

**Reply by ndonkoHenri** [reply]
Talking about what?
I gave you links right?

**Reply by 3mora2** [reply]
please add [This method could be used with page.window_prevent_close = True to implement app exit confirmation](https://flet.dev/docs/controls/page/#window_destroy) also with [on_window_event](https://flet.dev/docs/controls/page/#on_window_event) in documentation

**Comment by torablien** [comment]
hey @FeodorFitsner @ndonkoHenri, is it expected that [`Page.on_window_event`](https://flet.dev/docs/controls/page#on_window_event) only emits the `close` event if `page.window_prevent_close = True`? 

It feels a bit confusing to set `window_prevent_close = True` if you don't actually want to prevent close, but just listen for the close event so you can do something else (e.g. gracefully close threads). I initially thought it was a bug but wanted to check if this was behavior you intended before opening an issue. Thanks!

**Comment by dd-rongfa** [comment]
So what’s the finally anwser？

**Comment by ez404-36** [comment]
В документации flet (я использую 0.28.3) сказано сделать так:
```python

def on_event(e):
    if e.data == 'close':
       # bla-bla-bla your code before closing app
       page.window.destroy()

page.window.prevent_close = True
page.window.on_event = on_event
```
Это работает, но есть нюанс - процесс закрытия приложения занимает около 10 секунд, что не позволительно.
В моем случае мне надо было всего лишь сделать пару записей в БД перед закрытием, 
и проблема решилась следующим способом:
```python
page.on_disconnect = on_disconnect # coroutine

async def on_disconnect(e):
    # your awesome code (will be executed after closing flet window)
```
